src/studio/src/designer/backend/Controllers/DatamodelsController.cs

```diff
@@ -268,6 +284,24 @@ public ActionResult<IEnumerable<AltinnCoreFile>> GetDatamodels(string org, strin
             return Ok(schemaFiles);
         }
 
+        /// <summary>
+        /// Method that returns all xsd models within repository.
+        /// </summary>
+        /// <param name="org">the org owning the models repo</param>
+        /// <param name="repository">the model repos</param>
+        [Authorize]
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status302Found)]
+        [Route("xsd")]
+        public ActionResult<IEnumerable<AltinnCoreFile>> GetXSDDatamodels(string org, string repository)
```

Comment:
It'd be nice to add a integration test for new endpoint in DesignerTests project in Controllers/DataModelsController